torch/csrc/jit/codegen/cuda/dispatch.cpp

```diff
@@ -271,6 +271,8 @@ Statement* Statement::mutatorDispatch(T mutator, Statement* stmt) {
  * Handler template instantiations. These should only have to be done on base
  * classes. Actual visitors/mutators should inhereit from these classes and call
  * ->dispatch(this) to avoid needing an explicit instantiation.
+ * 
+ * NOTE - this doesn't quite work
```

Comment:
This comment was mostly for myself to look more into it.

Basically what I mean is that these specializations require exactly one of the known visitor base types (ex. OptOutDispatch*). You can't use a concrete visitor type with dispatch->(this) as the comment suggests, since there's no specialization for the concrete visitor type and we'd fail at link time. 

It's not really a bug - looking at the current visitors, the idiom seems to involve deferring explicitly to the visitor method:

  virtual void handle(const Statement* const s) {
    OptInConstDispatch::handle(s);
  };

I think this could be simplified though, thus the reminder comment.